File: sometest/src/main/java/com/kxs109/sometest/adapter/AreaSelectPageAdapter.java
package com.kxs109.sometest.adapter;

import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import java.util.List;

public class AreaSelectPageAdapter extends PagerAdapter {
    private List<View> mListView;

    public AreaSelectPageAdapter(List<View> listView) {
        this.mListView = listView;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        container.addView(mListView.get(position));
        return mListView.get(position);
    }

    @Override
    public int getCount() {
        return mListView.size();
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView(mListView.get(position));
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view ==object;
    }
}

File: lib_commonView/src/main/java/com/kxs109/commonview/areaSelect/adapter/AreaSelectPageAdapter.kt
package com.kxs109.commonview.areaSelect.adapter

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
 /**
  * @Description:    省市区pageAdapter
  * @Author:         zhh
  * @CreateDate:     2020/4/28 17:33
  */
class AreaSelectPageAdapter(private val mListView: List<View>) : PagerAdapter() {

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(mListView[position])
        return mListView[position]
    }

    override fun getCount(): Int {
        return mListView.size
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mListView[position])
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }
}
